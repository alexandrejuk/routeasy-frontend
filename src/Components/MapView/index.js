import React from 'react'
import {
  Map,
  Popup,
  TileLayer,
} from 'react-leaflet'
import 'leaflet/dist/leaflet.css'
import Marker from 'react-leaflet-enhanced-marker'
import PropTypes from 'prop-types'

import styles from './style.module.css'

const tileUrl = "https://cartodb-basemaps-{s}.global.ssl.fastly.net/light_all/{z}/{x}/{y}.png"
const initialConfig = {
  center: {
    lat: -23.531,
    lng: -46.584,
  },
  zoom: 12,
}

const MapView = ({
  deliveries,
}) => {
  const MarkerPin = index => (
    <div className={styles.marker}>
      <span className={styles.markerText}>
        {String(index + 1)}
      </span>
    </div>
  )

  const MarkerComponent = (delivery, index) => (
    <Marker
      key={index}
      icon={MarkerPin(index)}
      position={[delivery.lat, delivery.lng]}
    >
      <Popup>
        {delivery.name}<br />
        {delivery.weight} kg
      </Popup>
    </Marker>
  )

  return (
    <div className={styles.container}>
      <Map {...initialConfig}>
        <TileLayer url={tileUrl} />
        {deliveries && deliveries.map(MarkerComponent)}
      </Map>
    </div>
  )
}

MapView.propTypes = {
  deliveries: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      lat: PropTypes.number.isRequired,
      lng: PropTypes.number.isRequired,
      weight: PropTypes.number.isRequired,
    }).isRequired,
  ).isRequired,
}

export default MapView
