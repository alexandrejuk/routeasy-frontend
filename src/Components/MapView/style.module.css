@import '../../assets/css/colors.css';

.marker {
  background-color: var(--mine-shaft);
  border-radius: 50% 50% 50% 0;
  box-sizing: border-box;
  color: var(--white);
  font-size: 1.4em;
  font-weight: 600;
  height: 50px;
  padding: 10px;
  position: absolute;
  transform: perspective(40px) rotateX(20deg) rotateZ(-45deg);
  transform-origin: 50% 50%;
  width: 50px;
}

.markerText {
  display: block;
  text-align: center;
  transform: rotateY(0deg) rotateX(0deg) rotateZ(46deg);
}

.container {
  border: 1px solid var(--mine-shaft);
  border-radius: 5px;
  height: 550px;
  width: 100%;
}
