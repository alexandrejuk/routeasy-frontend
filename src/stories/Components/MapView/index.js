import React from 'react'
import { MapView } from '../../../Components'

export default {
  title: 'Components/MapView',
  component: MapView,
}

const Template = ({ data }) => (
  <MapView deliveries={data} />
)

export const Default = Template.bind({})

Default.args = {
  data: [{
    name: 'Eduardo',
    street: 'R. Ática',
    city: 'São Paulo',
    country: 'Brasil',
    lat: -23.531,
    lng: -46.584,
    weight: 10,
  }],
}
