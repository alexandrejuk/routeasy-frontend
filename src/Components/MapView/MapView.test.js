import React from 'react'
import { render } from '@testing-library/react'
import MapView from './'

const mockDelivery = [{
  name: 'Eduardo',
  street: 'R. Ática',
  city: 'São Paulo',
  country: 'Brasil',
  lat: -23.531,
  lng: -46.584,
  weight: 10,
}]

describe('MapView component', () => {
  it('should render correctly without marker', () => {
    const { container } = render(
      <MapView deliveries={[]} />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly one marker', () => {
    const { container, getByText } = render(
      <MapView deliveries={mockDelivery} />
    )
    getByText('1')
    expect(container).toMatchSnapshot()
  })
})
