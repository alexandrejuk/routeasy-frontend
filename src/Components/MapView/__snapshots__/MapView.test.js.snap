// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapView component should render correctly one marker 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://cartodb-basemaps-a.global.ssl.fastly.net/light_all/12/1517/2323.png"
                style="width: 256px; height: 256px; left: -250px; top: -151px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        >
          <div
            class="leaflet-marker-icon dummy leaflet-zoom-hide leaflet-interactive"
            style="margin-left: -50px; margin-top: -25px; width: 100px; height: 50px; left: 0px; top: 0px; z-index: 0;"
            tabindex="0"
          >
            
    
            <div
              style="zoom:100%;overflow:visible"
            >
              
      
              <div
                class="marker"
                data-reactroot=""
              >
                <span
                  class="markerText"
                >
                  1
                </span>
              </div>
              
    
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`MapView component should render correctly without marker 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="leaflet-container leaflet-grab leaflet-touch-drag"
      style="position: relative;"
      tabindex="0"
    >
      <div
        class="leaflet-pane leaflet-map-pane"
        style="left: 0px; top: 0px;"
      >
        <div
          class="leaflet-pane leaflet-tile-pane"
        >
          <div
            class="leaflet-layer "
            style="z-index: 1;"
          >
            <div
              class="leaflet-tile-container leaflet-zoom-animated"
              style="z-index: 18; left: 0px; top: 0px;"
            >
              <img
                alt=""
                class="leaflet-tile"
                role="presentation"
                src="https://cartodb-basemaps-a.global.ssl.fastly.net/light_all/12/1517/2323.png"
                style="width: 256px; height: 256px; left: -250px; top: -151px;"
              />
            </div>
          </div>
        </div>
        <div
          class="leaflet-pane leaflet-shadow-pane"
        />
        <div
          class="leaflet-pane leaflet-overlay-pane"
        />
        <div
          class="leaflet-pane leaflet-marker-pane"
        />
        <div
          class="leaflet-pane leaflet-tooltip-pane"
        />
        <div
          class="leaflet-pane leaflet-popup-pane"
        />
      </div>
      <div
        class="leaflet-control-container"
      >
        <div
          class="leaflet-top leaflet-left"
        >
          <div
            class="leaflet-control-zoom leaflet-bar leaflet-control"
          >
            <a
              aria-label="Zoom in"
              class="leaflet-control-zoom-in"
              href="#"
              role="button"
              title="Zoom in"
            >
              +
            </a>
            <a
              aria-label="Zoom out"
              class="leaflet-control-zoom-out"
              href="#"
              role="button"
              title="Zoom out"
            >
              −
            </a>
          </div>
        </div>
        <div
          class="leaflet-top leaflet-right"
        />
        <div
          class="leaflet-bottom leaflet-left"
        />
        <div
          class="leaflet-bottom leaflet-right"
        >
          <div
            class="leaflet-control-attribution leaflet-control"
          >
            <a
              href="https://leafletjs.com"
              title="A JS library for interactive maps"
            >
              Leaflet
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
